test/Dialect/FIRRTL/annotations.fir

```diff
@@ -339,50 +339,6 @@ circuit Foo: %[[{"a":null,"target":"~Foo|Foo>w[9]"}]]
 
 ; // -----
 
-; Annotations should apply even when the target's name is dropped.
```

Comment:
Since the parser doesn't determine if a name is droppable anymore, I don't think there is any need for this test anymore.  Please let me know if its actually still useful.